cli/metadata.ts

```diff
@@ -0,0 +1,55 @@
+#! /usr/bin/env node
+import { uploadRMRKMetadata } from "../src/tools/metadata-to-ipfs";
+import arg from "arg";
+import fs from "fs";
+import { NFTMetadata } from "../src/rmrk1.0.0/classes/nft";
+
+const fsPromises = fs.promises;
+
+interface MetadataSeedItem {
+  metadataFields: NFTMetadata;
+  imagePath?: string;
+}
+
+const metadata = async () => {
+  const args = arg({
+    "--input": String, // metadata input file
+    "--output": String, // metadata output file
+  });
+
+  const inputFile = await fsPromises.readFile(
+    args["--input"] || "./metadata-seed.example.json"
+  );
+  const metadataInput = JSON.parse(inputFile.toString());
+
+  if (!metadataInput?.metadata || !Array.isArray(metadataInput?.metadata)) {
+    throw new Error("Metadata json file missing 'metadata' array");
+  }
+
+  const promises = (metadataInput.metadata as MetadataSeedItem[]).map(
+    async (metadata, index) => {
+      if (!metadata.imagePath) {
```

Comment:
Also check if image is readable before attempting upload or doing anything else. In fact, it might be good to go through all the imagePath entries in the JSON first and cancel whole process if even one of them is missing / unreadable, so warn the user who's about to upload a ton of meta.

Reply:
Added validation